lanelet2/lanelet2_core/src/LaneletMap.cpp

```diff
@@ -412,6 +446,42 @@ void PrimitiveLayer<RegulatoryElementPtr>::add(const PrimitiveLayer<RegulatoryEl
   tree_->insert(element);
 }
 
+template <typename T>
+void PrimitiveLayer<T>::remove(Id id) {
+  // find the element with this id (the user must make sure it exists)
+  T element = elements_.find(id)->second;
+  
+  // remove from usage lookup
+  for (auto it = tree_->usage.ownedLookup.begin(); it != tree_->usage.ownedLookup.end(); )
+  {
+    if (it->second.id() == element.id()) { 
+      tree_->usage.ownedLookup.erase(it++); 
```

Comment:
Same question as above. Maybe I'm missing something about how erase works?

Reply:
My understanding is that post-increment iterator returns the iterator BEFORE the assignment. So iterator is increased, but it's returning the old one, which is why this works